receiver/statsdreceiver/transport/server_test.go

```diff
@@ -88,6 +102,27 @@ func Test_Server_ListenAndServe(t *testing.T) {
 			err = gc.Disconnect()
 			assert.NoError(t, err)
 
+			// Keep trying until we're timed out or got a result
+			timeout := time.After(10 * time.Second)
+			ticker := time.Tick(500 * time.Millisecond)
+			for {
+				stop := false
+				select {
+				// Got a timeout!
+				case <-timeout:
+					stop = true
+				// Got a tick
+				case <-ticker:
+					if len(transferChan) > 0 {
+						result = true
+					}
+				}
+				if stop {
+					break
+				}
+			}
```

Comment:
Use `assert.Eventually` https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually?

Reply:
Indeed, 
Updated PR to simplify using assert.Eventually